client/ayon_maya/plugins/load/load_reference.py

```diff
@@ -215,7 +215,15 @@ def process_reference(self, context, name, namespace, options):
                     self._set_display_handle(group_name)
 
             if product_type == "rig":
-                self._post_process_rig(namespace, context, options)
+                lock_set_on_load = settings['maya']['load'].get(
+                    'reference_loader', {}
+                ).get('lock_set_on_load', False)
+                self._post_process_rig(
+                    namespace,
+                    context,
+                    options,
+                    lock_set_on_load=lock_set_on_load
+                )
```

Comment:
Why not just pass the `settings`? That would also allow more data to flow through down the line?